### PR TITLE
Enable passing extra template paths [continued]

### PR DESCRIPTION
**Pull request by dajose**
This is #696 continuation...

@hackebrot sorry, I made a mess trying to bring #696 up to date with master and I ended up with a new PR.

(I am still not sure if somehow I could have just updated the branch to mantain the same PR)

### CODE REVIEW

**Comment by hackebrot** [comment]
Thank you @dajose! I will have a look as soon as I have some free time. 🙇

**Comment by theodesp** [comment]
@dajose I think you need to fix the conflicts first 👍

**Comment by dajose** [comment]
@theodesp done

**Comment by theodesp** [comment]
Cool, could you PM the rest of the team on Gitter about your progress? 👍 

https://gitter.im/audreyr/cookiecutter

File: tests/test_generate_files.py [hunk 1]
```diff
@@ -334,3 +334,21 @@ def test_raise_undefined_variable_project_dir(tmpdir):
     assert error.context == {}
 
     assert not output_dir.join('testproject').exists()
+
+
+def test_external_templates(tmpdir):
+    output_dir = generate.generate_files(
```
Comment (michaeljoseph) [inline tests/test_generate_files.py]: `output_dir` is a little confusing in this context, could we rename this to `project_dir` instead?

File: tests/test_generate_files.py [hunk 2]
```diff
@@ -334,3 +334,21 @@ def test_raise_undefined_variable_project_dir(tmpdir):
     assert error.context == {}
 
     assert not output_dir.join('testproject').exists()
+
+
```
Comment (michaeljoseph) [inline tests/test_generate_files.py]: Could we also have a test where `extra_templates` is a list?

File: cookiecutter/generate.py [hunk 3]
```diff
@@ -11,9 +11,11 @@
 import shutil
 from collections import OrderedDict
 
-from binaryornot.check import is_binary
 from jinja2 import FileSystemLoader
 from jinja2.exceptions import TemplateSyntaxError, UndefinedError
+from binaryornot.check import is_binary
```
Comment (michaeljoseph) [inline cookiecutter/generate.py]: This import should not have moved.

**Comment by dajose** [comment]
@michaeljoseph thanks for the feedback, changes applied :)

**Pull request closed without merging**